npm/src/clientfileprocessing/index.ts

```diff
@@ -23,7 +23,21 @@ export class ClientFileProcessing {
     this.s3Upload = s3Upload
   }
 
-  progressCallback(progressInformation: IProgressInformation) {
+  /**
+   * Updates a progress bar with a given percentage
+   * @param percent The percentage to update the progress bar with  of type {@link Number}
+   */
+  updateProgressBarPercentage(percent: Number) {
```

Comment:
I would refactor this function slightly to something like this. This can be reused as the call back for the different stages of the upload process:
1. Client file metadata upload
2. File upload

```
updateProgressBar(progressInformation: IProgressInformation) {
    const progressBarElement: HTMLDivElement | null = document.querySelector(
      ".progress-display"
    )

    if (progressBarElement) {
      progressBarElement.setAttribute("value", progressInformation.percentageProcessed.toString())
    }
  }
```